vendor/bat-native-ads/src/bat/ads/internal/processors/contextual/text_embedding/text_embedding_processor.cc

```diff
@@ -0,0 +1,60 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ads/internal/processors/contextual/text_embedding/text_embedding_processor.h"
+
+#include <iostream>
+#include <algorithm>
+
+#include "base/check.h"
+#include "bat/ads/internal/base/logging_util.h"
+#include "bat/ads/internal/ml/data/vector_data.h"
+#include "bat/ads/internal/deprecated/client/client_state_manager.h"
+#include "bat/ads/internal/ml/pipeline/text_processing/embedding_processing.h"
+#include "bat/ads/internal/resources/contextual/text_embedding/text_embedding_resource.h"
+#include "bat/ads/internal/serving/targeting/models/contextual/text_embedding/text_embedding_aliases.h"
+
+namespace ads {
+namespace processor {
+
+TextEmbedding::TextEmbedding(resource::TextEmbedding* resource)
+    : resource_(resource) {
+  DCHECK(resource_);
+}
+
+TextEmbedding::~TextEmbedding() = default;
+
+void TextEmbedding::Process(const std::string& text) {
+  if (!resource_->IsInitialized()) {
+    BLOG(1,
+         "Failed to process token embeddings as resource "
+         "not initialized");
+    return;
+  }
+
+  ml::pipeline::EmbeddingProcessing* embedding_proc_pipeline = resource_->get();
+  ml::VectorData text_embedding = embedding_proc_pipeline->EmbedText(text);
+
+  if (text_embedding.VectorSumElements() == 0.0) {
```

Comment:
embeddings elements can be negative, so this check might actually throw some false positives. Might be good in general to have a `IsNullVector` (or smth similar) function in `vector_data`

Reply:
Added GetNonZeroElementsCount() method